### PR TITLE
Initial support for NUCLEO-U5A5ZJ-Q board

### PR DESCRIPTION
**Pull request by kcheshmedzhiev**
## Summary
This PR adds support for stm32u5 NUCLEO-U5A5ZJ-Q board
## Impact
stm32u5 files in arch/arm/include/stm32u5 and arch/arm/src/stm32u5 directories. All files containing stm323u585 in their names have been renamed to stm32u5xx. According to reference manual RM0456 v4 the mcu line contains four major families with same cpu architecture and different memory and peripherals.  So, I think it is correct to have only one file for register description, memory map, etc. for entire line of microcontrollers.
Changes made also to boards/Kconfig and arch/arm/src/stm32u5/Kconfig.

## Testing
Here is a result of running a version with changes on NUCLEO-U5A5ZJ board

ABCDF

NuttShell (NSH) NuttX-12.2.1
nsh> uname -a
NuttX 12.2.1 f9386282dc-dirty Jul 18 2023 19:23:42 arm nucleo-u5a5zj-q
nsh> free
                   total       used       free    largest  nused  nfree
        Umem:    2548600       7792    2540808     851952     42      4
nsh>

and on B-U585I-IOT02A board because changes made to stm32u5 files

ABCDF

NuttShell (NSH) NuttX-12.2.1
nsh> uname -a
NuttX 12.2.1 f9386282dc-dirty Jul 18 2023 19:18:54 arm b-u585i-iot02a
nsh> free
                   total       used       free    largest  nused  nfree
        Umem:     189164       7716     181448     181448     36      1
nsh>

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@kcheshmedzhiev please rebase your change to the last master, macOS ci brokein was fixed.

**Comment by kcheshmedzhiev** [comment]
I did it... I think so.

**Comment by xiaoxiang781216** [comment]
Please run "./tools/refresh.sh --silent b-u585i-iot02a/nsh" to fix the ci error:
```
56d55
< CONFIG_STM32U5_SRAM1=y
58d56
< CONFIG_STM32U5_SRAM4=y
Saving the new configuration file
HEAD detached at pull/9853/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   boards/arm/stm32u5/b-u585i-iot02a/configs/nsh/defconfig
```

**Comment by kcheshmedzhiev** [comment]
It's done

**Comment by kcheshmedzhiev** [comment]
It's done

File: arch/arm/Kconfig [hunk 1]
```diff
@@ -523,6 +523,7 @@ config ARCH_CHIP_STM32U5
 	select ARM_HAVE_DSP
 	select ARCH_HAVE_FETCHADD
 	select ARCH_HAVE_HEAPCHECK
+  select ARCH_HAVE_HEAP2
```
Comment (xiaoxiang781216) [inline arch/arm/Kconfig:526]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 2]
```diff
@@ -20,10 +20,29 @@ config ARCH_CHIP_STM32U585AI
 	---help---
 		STM32 U5 Cortex M33, 2048 Kb FLASH, 768 Kb SRAM
 
-endchoice # STM32 L5 Chip Selection
+config ARCH_CHIP_STM32U5A5ZJT
+	bool "STM32U5A5ZJT"
+   select STM32U5_STM32U5A5XX
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:25]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 3]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
 config STM32U5_I2C1
 	bool "I2C1"
 	default n
+   select STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:501]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 4]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
 
 config STM32U5_I2C2
 	bool "I2C2"
 	default n
+   select STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:506]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 5]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
-	bool "CRS"
+config STM32U5_I2C3
+	bool "I2C3"
 	default n
+   select STM32U5_I2C
 
 config STM32U5_I2C4
 	bool "I2C4"
 	default n
+   select STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:516]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 6]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
 	bool "I2C2"
 	default n
+   select STM32U5_I2C
 
-config STM32U5_CRS
-	bool "CRS"
+config STM32U5_I2C3
+	bool "I2C3"
 	default n
+   select STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:511]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 7]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
 
 config STM32U5_I2C4
 	bool "I2C4"
 	default n
+   select STM32U5_I2C
+   
+config STM32U5_I2C5
+	bool "I2C5"
+   depends on STM32U5_STM32U59XX || STM32U5_STM32U59AXX || STM32U5_STM32U5A5XX || STM32U5_STM32U5A9XX
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:520]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 8]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
+	default n
+   select STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:522]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 9]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
+   
+config STM32U5_I2C6
+	bool "I2C6"
+   depends on STM32U5_STM32U59XX || STM32U5_STM32U59AXX || STM32U5_STM32U5A5XX || STM32U5_STM32U5A9XX
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:526]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 10]
```diff
@@ -457,18 +498,38 @@ config STM32U5_UART5
...
+	default n
+   select STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:528]: tab

File: arch/arm/src/stm32u5/Kconfig [hunk 11]
```diff
@@ -3171,13 +3223,17 @@ config STM32U5_SPI_DMA
...
+	default n
+
 menu "I2C Configuration"
 	depends on STM32U5_I2C
 
 config STM32U5_I2C_DYNTIMEO
 	bool "Use dynamic timeouts"
 	default n
-	depends on STM32U5_I2C
+# depends on STM32U5_I2C
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:3236]: why comment out

File: arch/arm/src/stm32u5/Make.defs [hunk 12]
```diff
@@ -54,6 +54,6 @@ endif
 
 # Required chip type specific files
 
-ifeq ($(CONFIG_STM32U5_STM32U585XX),y)
-CHIP_CSRCS += stm32u585xx_rcc.c
-endif
+ifeq ($(CONFIG_USBDEV),y)
+CHIP_CSRCS += stm32_otgdev.c
+endif
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Make.defs:59]: add blank line

File: arch/arm/src/stm32u5/hardware/stm32_flash.h [hunk 13]
```diff
@@ -41,6 +41,7 @@
  *   Parts STM32U585 and STM32U575 have 2048Kb of FLASH
  */
 
+#if defined(CONFIG_ARCH_CHIP_STM32U585AI)
 #if !defined(CONFIG_STM32U5_FLASH_OVERRIDE_DEFAULT) && \
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/hardware/stm32_flash.h:45]:
```suggestion
#if  defined(CONFIG_ARCH_CHIP_STM32U585AI) && \
      !defined(CONFIG_STM32U5_FLASH_OVERRIDE_DEFAULT) && \
```

File: arch/arm/src/stm32u5/hardware/stm32_flash.h [hunk 14]
```diff
@@ -70,11 +71,18 @@
...
 #  error "unknown flash configuration!"
 #endif
+#endif
 
 #ifdef STM32_FLASH_PAGESIZE
 #  define STM32_FLASH_SIZE        (STM32_FLASH_NPAGES * STM32_FLASH_PAGESIZE)
 #endif
 
+#if defined(CONFIG_STM32U5_STM32U5A5XX)
+#define STM32_FLASH_NPAGES             512
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/hardware/stm32_flash.h:81]:
```suggestion
#  define STM32_FLASH_NPAGES     512
```

File: arch/arm/src/stm32u5/hardware/stm32_memorymap.h [hunk 15]
```diff
@@ -43,12 +43,35 @@
 
 /* Code Base Addresses ******************************************************/
 
+#if defined(CONFIG_STM32U5_STM32U535XX) || defined(CONFIG_STM32U5_STM32U545XX)
+#define STM32_BOOT_BASE         0x00000000  /* 0x00000000-0x000fffff: Aliased boot memory */
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/hardware/stm32_memorymap.h:47]:
```suggestion
#  define STM32_BOOT_BASE       0x00000000  /* 0x00000000-0x000fffff: Aliased boot memory */
```

File: arch/arm/src/stm32u5/stm32_allocateheap.c [hunk 16]
```diff
@@ -350,6 +357,26 @@ void arm_addregion(void)
 
 #endif /* SRAM3 */
 
+#ifdef CONFIG_STM32U5_SRAM5_HEAP
+
+#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/stm32_allocateheap.c:362]:
```suggestion
#  if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
```

File: boards/Kconfig [hunk 17]
```diff
@@ -3106,6 +3114,7 @@ config ARCH_BOARD
 	default "b-l072z-lrwan1"            if ARCH_BOARD_B_L072Z_LRWAN1
 	default "b-l475e-iot01a"            if ARCH_BOARD_B_L475E_IOT01A
 	default "b-u585i-iot02a"            if ARCH_BOARD_B_U585I_IOT02A
+  default "nucleo-u5a5zj-q"           if ARCH_BOARD_NUCLEO_U5A5ZJ_Q
```
Comment (xiaoxiang781216) [inline boards/Kconfig:3117]:
```suggestion
	default "nucleo-u5a5zj-q"           if ARCH_BOARD_NUCLEO_U5A5ZJ_Q
```

File: boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+    #include <nuttx/timers/ds3231.h>
+#endif /* CONFIG_RTC_DSXXXX */
+
+#endif /* CONFIG_I2C */
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define DEVNO_ZERO				0
+#define DEVNO_ONE					1
```
Comment (xiaoxiang781216) [inline boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c:58]: change tab to space and align

**Comment by kcheshmedzhiev** [comment]
Changed typos. File stm32_flash.c (as well as other files like stm32_pinmap.h, stm32_memorymap.h, etc.) need more precise review. So I will edit them in next round.

File: boards/Kconfig [hunk 19]
```diff
@@ -3114,7 +3114,7 @@ config ARCH_BOARD
 	default "b-l072z-lrwan1"            if ARCH_BOARD_B_L072Z_LRWAN1
 	default "b-l475e-iot01a"            if ARCH_BOARD_B_L475E_IOT01A
 	default "b-u585i-iot02a"            if ARCH_BOARD_B_U585I_IOT02A
-  default "nucleo-u5a5zj-q"           if ARCH_BOARD_NUCLEO_U5A5ZJ_Q
+	default "nucleo-u5a5zj-q"						if ARCH_BOARD_NUCLEO_U5A5ZJ_Q
```
Comment (xiaoxiang781216) [inline boards/Kconfig:3117]: remove the extra space before ARCH_BOARD_NUCLEO_U5A5ZJ_Q

File: boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c [hunk 20]
```diff
@@ -46,8 +46,8 @@
 FAR struct i2c_master_s *i2c1_m;
 FAR struct i2c_master_s *i2c2_m;
 #ifdef CONFIG_RTC_DSXXXX
-	#include <nuttx/timers/rtc.h>
-	#include <nuttx/timers/ds3231.h>
+#	include <nuttx/timers/rtc.h>
```
Comment (xiaoxiang781216) [inline boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c:49]: remove the extra space

**Comment by xiaoxiang781216** [comment]
> Changed typos. File stm32_flash.c (as well as other files like stm32_pinmap.h, stm32_memorymap.h, etc.) need more precise review. So I will edit them in next round.

@kcheshmedzhiev not all comment get addressed.

**Comment by kcheshmedzhiev** [comment]
Just double checked board/config and  boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c and found no extra spaces. Would you point me how to resolve the issue(s)?

**Comment by xiaoxiang781216** [comment]
> Just double checked boards/Kconfig and boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c and found no extra spaces. Would you point me how to resolve the issue(s)?

Please check the review comment without "Outdated" label:
image

**Comment by kcheshmedzhiev** [comment]
I made changes.

**Comment by kcheshmedzhiev** [comment]
One question: What utility you use to check sources? I ran tools/checkpatch.sh and it didn't show me any errors.

**Comment by xiaoxiang781216** [comment]
> One question: What utility you use to check sources? I ran tools/checkpatch.sh and it didn't show me any errors.

Sorry, checkpatch.sh can't find all style violation. Most comment comes from the current code base and this link:
https://nuttx.apache.org/docs/latest/contributing/coding_style.html

**Comment by xiaoxiang781216** [comment]
> I made changes.

But the indent is two spaces not four.

File: arch/arm/include/stm32u5/stm32u5xx_irq.h [hunk 21]
```diff
@@ -128,7 +128,15 @@
 #define STM32_IRQ_TIM16         (STM32_IRQ_FIRST + 70)  /* 70:  TIM16 global interrupt */
 #define STM32_IRQ_TIM17         (STM32_IRQ_FIRST + 71)  /* 71:  TIM17 global interrupt */
 #define STM32_IRQ_COMP          (STM32_IRQ_FIRST + 72)  /* 72:  COMP1/COMP2 interrupts */
-#define STM32_IRQ_OTG_FS        (STM32_IRQ_FIRST + 73)  /* 73:  USB OTG FS global interrupt */
+#if defined(CONFIG_STM32U5_STM32U535XX) || defined(CONFIG_STM32U5_STM32U545XX) || \
+    defined(CONFIG_STM32U5_STM32U575XX) || defined(CONFIG_STM32U5_STM32U585XX)
+# define STM32_IRQ_OTG_HS        (STM32_IRQ_FIRST + 73)  /* 73:  USB OTG FS global interrupt */
+#elif defined(CONFIG_STM32U5_STM32U59XX) || defined(CONFIG_STM32U5_STM32U59AXX)  || \
+    defined(CONFIG_STM32U5_STM32U5A5XX) || defined(CONFIG_STM32U5_STM32U5A9XX)
+# define STM32_IRQ_OTG_HS        (STM32_IRQ_FIRST + 73)  /* 73:  USB OTG HS global interrupt */
```
Comment (xiaoxiang781216) [inline arch/arm/include/stm32u5/stm32u5xx_irq.h:136]:
```suggestion
#  define STM32_IRQ_OTG_HS       (STM32_IRQ_FIRST + 73)  /* 73:  USB OTG HS global interrupt */
```

File: arch/arm/src/stm32u5/Kconfig [hunk 22]
```diff
@@ -284,7 +275,63 @@ config STM32U5_DCACHE1
...
+	default n
+	depends on STM32U5_STM32U575XX || STM32U5_STM32U585XX || STM32U5_STM32U59XX || STM32U5_STM32U59AXX || \
+              STM32U5_STM32U5A5XX || STM32U5_STM32U5A9XX   
+
+comment "SRAM Options"
+
+config STM32U5_SRAM2_HEAP
+	bool "SRAM2 is used for heap"
 	default n
+   depends on STM32U5_SRAM2
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:301]:
```suggestion
	depends on STM32U5_SRAM2
```

File: arch/arm/src/stm32u5/Kconfig [hunk 23]
```diff
@@ -284,7 +275,63 @@ config STM32U5_DCACHE1
...
+		which may be used by the application for special purposes.  But if these
+		special properties are not needed, it may be instead added to the heap for
+		use by malloc().
+		NOTE: you must also select an appropriate number of memory regions in the
+		'Memory Management' section.
+      
+config STM32U5_SRAM2_INIT
+	bool "SRAM2 is initialized to zero"
+	default n
+   depends on STM32U5_SRAM2
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:314]:
```suggestion
	depends on STM32U5_SRAM2
```

File: arch/arm/src/stm32u5/Kconfig [hunk 24]
```diff
@@ -284,7 +275,63 @@ config STM32U5_DCACHE1
...
+		time.      
+      
+config STM32U5_SRAM3_HEAP
+	bool "SRAM3 is used for heap"
+	depends on STM32U5_SRAM3
+	default n
+   
+config STM32U5_SRAM5_HEAP
+	bool "SRAM5 is used for heap"
+   depends on STM32U5_SRAM5
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:333]:
```suggestion
	depends on STM32U5_SRAM5
```

File: arch/arm/src/stm32u5/Kconfig [hunk 25]
```diff
@@ -296,8 +343,14 @@ config STM32U5_DCMI_PSSI
 	bool "DCMI_PSSI"
 	default n
 
-config STM32U5_OTG
-	bool "OTG"
+config STM32U5_OTGFS
+	bool "OTG FS"
+   depends on STM32U5_STM32U535XX || CONFIG_STM32U5_STM32U545XX || STM32U5_STM32U575XX || STM32U5_STM32U585XX
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:348]:
```suggestion
	depends on STM32U5_STM32U535XX || CONFIG_STM32U5_STM32U545XX || STM32U5_STM32U575XX || STM32U5_STM32U585XX
```

File: arch/arm/src/stm32u5/Kconfig [hunk 26]
```diff
@@ -296,8 +343,14 @@ config STM32U5_DCMI_PSSI
...
+	default n
+
+config STM32U5_OTGHS
+	bool "OTG HS"
+   depends on STM32U5_STM32U59XX || STM32U5_STM32U59AXX || STM32U5_STM32U5A5XX || STM32U5_STM32U5A9XX
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/Kconfig:353]:
```suggestion
	depends on STM32U5_STM32U59XX || STM32U5_STM32U59AXX || STM32U5_STM32U5A5XX || STM32U5_STM32U5A9XX
```

File: arch/arm/src/stm32u5/hardware/stm32_flash.h (hunk 13 above)
Comment (xiaoxiang781216) [reply arch/arm/src/stm32u5/hardware/stm32_flash.h:45]: Unchange, you need either merge two #if into one, or indent the second #if

File: arch/arm/src/stm32u5/hardware/stm32_flash.h [hunk 27]
```diff
@@ -70,11 +71,18 @@
...
 #  error "unknown flash configuration!"
 #endif
+#endif
 
 #ifdef STM32_FLASH_PAGESIZE
 #  define STM32_FLASH_SIZE        (STM32_FLASH_NPAGES * STM32_FLASH_PAGESIZE)
 #endif
 
+#if defined(CONFIG_STM32U5_STM32U5A5XX)
+#	define STM32_FLASH_NPAGES             512
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/hardware/stm32_flash.h:81]:
```suggestion
#  define STM32_FLASH_NPAGES             512
```
change all tab to space in *.h/*.c, tab is only in Kconfig and *.S.

File: arch/arm/src/stm32u5/hardware/stm32_memorymap.h [hunk 28]
```diff
@@ -43,11 +43,34 @@
 
 /* Code Base Addresses ******************************************************/
 
-#define STM32_BOOT_BASE         0x00000000  /* 0x00000000-0x000fffff: Aliased boot memory */
-#define STM32_FLASH_BASE        0x08000000  /* 0x08000000-0x081fffff: FLASH memory */
-#define STM32_SRAM1_BASE        0x20000000  /* 0x20000000-0x2002ffff: 192k SRAM1 */
-#define STM32_SRAM2_BASE        0x20030000  /* 0x20030000-0x2003ffff:  64k SRAM2 */
-#define STM32_SRAM3_BASE        0x20040000  /* 0x20040000-0x200bffff: 512k SRAM3 */
+#if defined(CONFIG_STM32U5_STM32U535XX) || defined(CONFIG_STM32U5_STM32U545XX)
+#	define STM32_BOOT_BASE         0x00000000  /* 0x00000000-0x000fffff: Aliased boot memory */
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/hardware/stm32_memorymap.h:47]:
```suggestion
#  define STM32_BOOT_BASE         0x00000000  /* 0x00000000-0x000fffff: Aliased boot memory */
```
again, don't use tab in source files, please enable showing special char in your editor

File: arch/arm/src/stm32u5/stm32_allocateheap.c [hunk 29]
```diff
@@ -350,6 +357,26 @@ void arm_addregion(void)
...
+
+  /* Allow user-mode access to the SRAM5 heap */
+
+  stm32_mpu_uheap((uintptr_t)SRAM5_START, STM32_SRAM5_SIZE);
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/stm32_allocateheap.c:366]:
```suggestion
stm32_mpu_uheap((uintptr_t)SRAM5_START, STM32_SRAM5_SIZE);
```

File: arch/arm/src/stm32u5/stm32_allocateheap.c [hunk 30]
```diff
@@ -350,6 +357,26 @@ void arm_addregion(void)
...
+
+#endif
+
+  /* Colorize the heap for debug */
+
+  up_heap_color((void *)SRAM5_START, STM32_SRAM5_SIZE);
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32u5/stm32_allocateheap.c:372]: remove the indent, only preprocess statement need the indent.

File: arch/arm/src/stm32u5/stm32_allocateheap.c (hunk 16 above)
Comment (xiaoxiang781216) [reply arch/arm/src/stm32u5/stm32_allocateheap.c:362]: unchange

File: boards/Kconfig (hunk 19 above)
Comment (xiaoxiang781216) [reply boards/Kconfig:3117]: not fixed

Comment (xiaoxiang781216) [reply boards/Kconfig:3117]:
```suggestion
	default "nucleo-u5a5zj-q"           if ARCH_BOARD_NUCLEO_U5A5ZJ_Q
```
AGAIN, DON'T USE TAB IN SOURCE CODE.

File: boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c (hunk 18 above)
Comment (xiaoxiang781216) [reply boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c:59]: no change

**Comment by kcheshmedzhiev** [comment]
I made corrections. I think so. I got lost in the world of tabs and spaces :) Sorry

File: arch/arm/src/stm32u5/stm32_allocateheap.c (hunk 30 above)
Comment (xiaoxiang781216) [reply arch/arm/src/stm32u5/stm32_allocateheap.c:372]: no change

File: arch/arm/src/stm32u5/stm32_allocateheap.c (hunk 16 above)
Comment (xiaoxiang781216) [reply arch/arm/src/stm32u5/stm32_allocateheap.c:362]: still no change

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@kcheshmedzhiev 

When you squash your commits, please amend the commit messages as well.

```
commit ea87d008a0dfe5b2516cf7655baa0c5c88d05ceb (HEAD -> master, upstream/master)
Author: kcheshmedzhiev <cheshmedzhiev@gmail.com>
Date:   Tue Jul 18 19:27:19 2023 +0300

    Initial support for NUCLEO-U5A5ZJ-Q board

    Fix for ci error

    Typo changes

    Fixed typo in boards/arm/stm32u5/nucleo-u5a5zj-q/src/stm32_bringup.c

    Typo changes

    Typo fixes

    Typo fixes

    Typo changes

    Typo changes

    Typo changes
```

**Pull request merged**